src/modules/projects/components/ProjectReferralPostingForm.tsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useState } from 'react';
+
+import { emptyErrorState, partitionValidations } from '@/modules/errors/util';
+import DynamicForm, {
+  DynamicFormOnSubmit,
+} from '@/modules/form/components/DynamicForm';
+import { ReferralPostingDefinition } from '@/modules/form/data';
+import { transformSubmitValues } from '@/modules/form/util/formUtil';
+import {
+  ReferralPostingDetailFieldsFragment,
+  ReferralPostingInput,
+  useUpdateReferralPostingMutation,
+} from '@/types/gqlTypes';
+
+interface Props {
+  referralPosting: ReferralPostingDetailFieldsFragment;
+}
+export const ProjectReferralPostingForm: React.FC<Props> = ({
+  referralPosting,
+}) => {
+  const [errors, setErrors] = useState(emptyErrorState);
+  const [mutate, { loading }] = useUpdateReferralPostingMutation({
+    onCompleted: (data) => {
+      const { errors: remoteErrors = [] } = data.updateReferralPosting || {};
+      if (remoteErrors.length) {
+        setErrors(partitionValidations(remoteErrors));
+      }
+    },
+    onError: (apolloError) => setErrors({ ...emptyErrorState, apolloError }),
+  });
+  const handleSubmit: DynamicFormOnSubmit = useCallback(
+    ({ values }) => {
+      const input = transformSubmitValues({
+        definition: ReferralPostingDefinition,
+        values,
+        keyByFieldName: true,
+      }) as ReferralPostingInput;
+
+      console.debug('submitting', input);
+      mutate({
+        variables: {
+          id: referralPosting.id,
+          input,
+        },
+      });
+    },
+    [mutate, referralPosting.id]
+  );
+
+  return (
+    <DynamicForm
+      definition={ReferralPostingDefinition}
+      FormActionProps={{
+        submitButtonText: 'Update Referral',
+        discardButtonText: 'Cancel',
+      }}
+      initialValues={referralPosting}
```

Comment:
Sooo - the posting itself can't just be passed in as `initialValues`, it needs to be transformed first. Definitely a confusing API. You'll need to call "createInitialValuesFromRecord" to create the initial form state which is keyed by link id. It also transforms the values (like creating date object, picklist object, etc)

Reply:
That should fix the issue you saw with DynamicField

Reply:
Got it thanks for explaining. I backed out the previous fix I made to the select component